src/rgw/rgw_rest.cc

```diff
@@ -447,21 +473,17 @@ void dump_uri_from_state(struct req_state *s)
       location += "/";
       if (!s->object.empty()) {
 	location += s->object.name;
-	STREAM_IO(s)->print("Location: %s\r\n", location.c_str());
+	dump_header(s, "Location", location);
       }
     }
-  }
-  else {
-    STREAM_IO(s)->print("Location: \"%s\"\r\n", s->info.request_uri.c_str());
+  } else {
+    dump_header(s, "Location", "\"" + s->info.request_uri + "\"");
```

Comment:
can we find a way to avoid `operator+()` on strings for stuff like this? maybe add a `dump_quoted_header()` function for cases like this and `lo_etag`?